scripts/background.js

```diff
@@ -75,47 +75,26 @@ chrome.webRequest.onCompleted.addListener(function(details) {
         isValidUrl(details.url)) {
               Globalstatuscode=details.statusCode;
               wmAvailabilityCheck(details.url, function(wayback_url, url) {
-              if(details.statusCode==504){
-                  //notify(wayback_url,'View an archived version courtesy of the Internet Archive WayBack Machine');
-                  chrome.notifications.create(
-                'wayback-notification',{   
-                type: 'basic', 
-                requireInteraction:true,
-                iconUrl: '/images/logo.gif', 
-                title: "Page not available ?", 
-                message:"View an archived version courtesy of the WayBack Machine",
-                buttons:[{
-                    title: "Click here to see archived version"
-                }]
-                },
-                function(id){
-                    myNotID=id;
-                } 
-              );
-              chrome.notifications.onButtonClicked.addListener(function(notifId, btnIdx) {
-    if (notifId === myNotID) {
-        if (btnIdx === 0) {
-                chrome.tabs.create({ url:wayback_url});
-                chrome.notifications.clear(myNotID);
-                myNotID=null;
-        }
-    }
-});
-              }else{
-                  chrome.tabs.executeScript(details.tabId, {
-              file: "scripts/client.js"
-            }, function() {
-              chrome.tabs.sendMessage(details.tabId, {
-                type: "SHOW_BANNER",
-                wayback_url: wayback_url,
-                page_url: details.url,
-                status_code: details.statusCode
+                try{     // On pages which return 504, 503 etc chrome.tabs.executeScript() throws an error because client.js cannot be injected
+                    chrome.tabs.executeScript(details.tabId, {
+                    file: "scripts/client.js"
+              },function() {
+                    chrome.tabs.sendMessage(details.tabId, {
+                    type: "SHOW_BANNER",
+                    wayback_url: wayback_url,
+                    page_url: details.url,
+                    status_code: details.statusCode
               });
             });
-              }
+                }
+                catch(error){
+                    chrome.tabs.update(details.tabId, {url: chrome.extension.getURL('dnserror.html')+"?wayback_url="+wayback_url+"?page_url="+url+"?status_code="+details.statusCode+"?"});
+                }
+                
+              
           }, function() {
             
-          });
+          });            
```

Comment:
Could you please avoid having unwanted trailing spaces here?
Tip: vim has many ways to *highlight trailing spaces*, please look it up, its possible to do it in all major editors AFAIK.